web-frontend/src/pages/Settings/SettingsSignRule.tsx

```diff
@@ -19,36 +28,57 @@ interface SettingsSignRulesProps {
 export default function SettingsSignRules(props: SettingsSignRulesProps) {
   const { nickname } = props;
 
-  const {
-    data: config,
-    isLoading: isConfigLoading,
-    error: configError,
-  } = useResource<Config>('config', true);
-
-  const { mutate: addSignRule, error: addSignRuleError } = usePost(
-    `accounts/${nickname}/signrules`,
+  const [editingRule, setEditingRule] = useState<SignRule | undefined>(
+    undefined,
   );
+  const [isAddEditRuleModalOpen, setIsAddEditRuleModalOpen] = useState(false);
+
+  const [config, setConfig] = useState<Config>();
+  const [isLoading, setIsLoading] = useState(true);
+
+  const msWallet = useMemo(() => new MassaStationWallet(), []);
+
+  const fetchConfig = async () => {
+    try {
+      const walletConfig = await msWallet.getConfig();
+      setConfig(walletConfig as Config);
```

Comment:
you should not cast.
better allow undefined in the useState
`useState<Config | undefined>();`